esm_runscripts/coupler.py

```diff
@@ -16,14 +16,16 @@ def __init__(self, full_config, name):
         if "coupling_target_fields" in full_config[self.name]:
             for restart_file in list(full_config[self.name]["coupling_target_fields"]):
                 self.nb_of_couplings += len(list(full_config[self.name]["coupling_target_fields"][restart_file]))
+        if full_config['fesom'].get("do_hosing"):
```

Comment:
I think this line can be reached from a model that has no `fesom` component and therefore, would lead to a crash. I would suggest to change this line to something like:
```
if full_config.get('fesom', {}).get("do_hosing"):
```
This is in principle a more general and safe solution.

Reply:
```suggestion
        if full_config.get('fesom', {}).get("do_hosing"):
```